lib/src/block_syntaxes/table_syntax.dart

```diff
@@ -71,29 +72,42 @@ class TableSyntax extends BlockSyntax {
   }
 
   List<String?> _parseAlignments(String line) {
-    final startIndex = _walkPastOpeningPipe(line);
+    final columns = <String?>[];
+    // Set the value to `true` when hitting a non whitespace character other
+    // than the first pipe character.
+    var started = false;
+    var hitDash = false;
+    String? align;
+
+    for (var i = 0; i < line.length; i++) {
+      final char = line.codeUnitAt(i);
+      if (char == $space || char == $tab || (!started && char == $pipe)) {
+        continue;
+      }
+      started = true;
 
-    var endIndex = line.length - 1;
-    while (endIndex > 0) {
-      final ch = line.codeUnitAt(endIndex);
-      if (ch == $pipe) {
-        endIndex--;
-        break;
+      if (char == $colon) {
+        if (hitDash) {
+          align = align == 'left' ? 'center' : 'right';
+        } else {
+          align = 'left';
+        }
       }
-      if (ch != $space && ch != $tab) {
-        break;
+
+      if (char == $pipe) {
+        columns.add(align);
+        hitDash = false;
+        align = null;
+      } else {
+        hitDash = true;
```

Comment:
Hmm, I think this variable maybe should be changed. So it doesn't necessarily mean, "I've hit a dash." It more means, "I've moved past the chance to hit an alignment dash." Is that right?

Reply:
I need something means now I am at the position of a dash ( when `final char = line.codeUnitAt(i);` is a dash)

It might not always move past a dash, it might end up on a dash when there is no ending pipe, for example
```
|--:|--
```

Reply:
The `hitDash` helps to determine if a colon is on the left `:---` or right `---:` or both side `:---:`

Reply:
OK I think I understand.